lib/Dialect/FIRRTL/Transforms/GrandCentral.cpp

```diff
@@ -1363,6 +1443,108 @@ bool GrandCentralPass::traverseField(
       .Default([](auto a) { return true; });
 }
 
+bool GrandCentralPass::traverseViewField(
+    Attribute field, VerbatimBuilder &path,
+    SmallVector<VerbatimXMRbuilder> &xmrElems,
+    SmallVector<InterfaceElemsBuilder> &interfaceBuilder, ViewIntrinsicOp view,
+    size_t &idx) {
+  return TypeSwitch<Attribute, bool>(field)
+      .Case<AugmentedGroundTypeAttr>([&](AugmentedGroundTypeAttr ground) {
+        // Grab next signal, increment index counter.
+        auto index = idx++;
+        if (index >= view.getNumOperands()) {
+          view.emitError("more ground types found than view has operands");
+          return false;
+        }
+
+        auto val = view.getOperand(index);
+        auto tpe = type_cast<FIRRTLBaseType>(val.getType());
+
+        // If the type is zero-width then do not emit an XMR.
+        if (!tpe.getBitWidthOrSentinel())
+          return true;
+
+        /// Increment all the indices inside `{{`, `}}` by one. This is to
+        /// indicate that a value is added to the `substitutions` of the
+        /// verbatim op, other than the symbols.
+        auto getStrAndIncrementIds = [&](StringRef base) -> StringAttr {
+          SmallString<128> replStr;
+          StringRef begin = "{{";
+          StringRef end = "}}";
+          // The replacement string.
+          size_t from = 0;
+          while (from < base.size()) {
+            // Search for the first `{{` and `}}`.
+            size_t beginAt = base.find(begin, from);
+            size_t endAt = base.find(end, from);
+            // If not found, then done.
+            if (beginAt == StringRef::npos || endAt == StringRef::npos ||
+                (beginAt > endAt)) {
+              replStr.append(base.substr(from));
+              break;
+            }
+            // Copy the string as is, until the `{{`.
+            replStr.append(base.substr(from, beginAt - from));
+            // Advance `from` to the character after the `}}`.
+            from = endAt + 2;
+            auto idChar = base.substr(beginAt + 2, endAt - beginAt - 2);
+            int idNum;
+            bool failed = idChar.getAsInteger(10, idNum);
+            (void)failed;
+            assert(!failed && "failed to parse integer from verbatim string");
+            // Now increment the id and append.
+            replStr.append("{{");
+            Twine(idNum + 1).toVector(replStr);
+            replStr.append("}}");
+          }
+          return StringAttr::get(&getContext(), "assign " + replStr + ";");
+        };
+
+        // This is the new style of XMRs using RefTypes.  The value substitution
+        // index is set to -1, as it will be incremented when generating the
+        // string.
+        // Generate the path from the LCA to the module that contains the leaf.
```

Comment:
The last sentence of this comment is, I think, now stale. (I think it's stale in the copied code, too, as there shouldn't be any LCA computation anymore.)